userspace/libsinsp/container.h

```diff
@@ -38,22 +38,71 @@ limitations under the License.
 class sinsp_container_manager
 {
 public:
-	using map_ptr_t = const std::unordered_map<std::string, sinsp_container_info>*;
-	using entry_ptr_t = sinsp_container_info*;
+	using map_ptr_t = const std::unordered_map<std::string, sinsp_container_info::ptr_t>*;
 
 	sinsp_container_manager(sinsp* inspector);
 	virtual ~sinsp_container_manager();
 
-	map_ptr_t get_containers();
+	/**
+	 * @brief Get the whole container map (read-only)
+	 * @return the map of container_id -> shared_ptr<container_info>
+	 */
+	map_ptr_t get_containers() const;
 	bool remove_inactive_containers();
-	void add_container(const sinsp_container_info& container_info, sinsp_threadinfo *thread);
-	entry_ptr_t get_container(const std::string &id);
+
+	/**
+	 * @brief Add/update a container in the manager map, executing on_new_container callbacks
+	 *
+	 * @param container_info shared_ptr owning the container_info to add/update
+	 * @param thread a thread in the container, only passed to callbacks
+	 */
+	void add_container(const sinsp_container_info::ptr_t& container_info, sinsp_threadinfo *thread);
+
+	/**
+	 * @brief Update a container by replacing its entry with a new one
```

Comment:
nice job on these

fwiw, I don't usually bother to list out params. I just care about the brief (on public functions) along with anything the client needs to know

I dunno what our official stance is